testsuite/cluster-test/src/prometheus.rs

```diff
@@ -100,9 +102,45 @@ impl Prometheus {
             .avg()
             .ok_or_else(|| format_err!("Failed to compute avg"))
     }
+
+    pub fn query_range_max(
+        &self,
+        query: String,
+        start: &Duration,
+        end: &Duration,
+        step: u64,
+    ) -> Result<HashMap<String, f64>> {
+        let response = self.query_range(query, start, end, step)?;
+        response
+            .max()
+            .ok_or_else(|| format_err!("Failed to compute max"))
+    }
 }
 
 impl MatrixResponse {
+    pub fn max(&self) -> Option<HashMap<String, f64>> {
+        if self.inner.is_empty() {
```

Comment:
If `self.inner` is empty, what would the length of the `max` `HashMap` be?

Reply:
I assume if self.inner is empty, it returns None? Do we need to care max HashMap then?

Reply:
If self.inner is empty, the iteration through its elements is a no-op, the max HashMap is empty, and you return None already.

Reply:
Sorry I think I did not get the point before. yes, this check here is redundant I will remove it, we will eventually return None if inner is empty